lib/Optimizer/Transforms/LowerUnwind.cpp

```diff
@@ -309,6 +307,20 @@ populateTerminatorAllocaMap(const SmallVector<Operation *> &terminators,
   return results;
 }
 
+static bool fullPathPrimitive(
```

Comment:
Maybe it's just me, but the name `fullPathPrimitive()` makes it sound like it's checking that _all_ parents up the chain are primitive, but the code looks like it's only checking that at least one ancestor is primitive. Perhaps a better name would be `anyPrimitiveAncestors()`?